### PR TITLE
applications: nrf_desktop: Keyboard multiple peer and bugfix

### PR DESCRIPTION
**Pull request by MarekPieta**
Change adds multiple peer support for keyboard (with new LED effect) and fixes bug with allocating too much memory for LED states.

### CODE REVIEW

File: applications/nrf_desktop/configuration/nrf52_pca20037/led_state_def.h [hunk 1]
```diff
@@ -26,12 +26,27 @@ static const u8_t led_map[LED_ID_COUNT] = {
 /* System state LED is unavailable - leave undefined */
 static const struct led_effect led_system_state_effect[LED_SYSTEM_STATE_COUNT];
 
-static const struct led_effect led_peer_state_effect[CONFIG_BT_MAX_PAIRED][LED_PEER_STATE_COUNT] = {
+static const struct led_effect led_peer_state_effect[CONFIG_BT_MAX_PAIRED - 1][LED_PEER_STATE_COUNT] = {
```
Comment (pdunaj) [inline applications/nrf_desktop/configuration/nrf52_pca20037/led_state_def.h]: maybe we can introduce a define (or a config in kconfig) that would give peer count?

File: applications/nrf_desktop/configuration/common/led_effect.h [hunk 2]
```diff
@@ -116,6 +117,35 @@ struct led_effect {
 		.loop_forever = true,						\
 	}
 
+
```
Comment (pdunaj) [inline applications/nrf_desktop/configuration/common/led_effect.h]: explain in the comment here why we cannot give period as argument

Comment (MarekPieta) [reply applications/nrf_desktop/configuration/common/led_effect.h]: done

File: applications/nrf_desktop/configuration/nrf52_pca20037/led_state_def.h (hunk 1 above)
Comment (MarekPieta) [reply applications/nrf_desktop/configuration/nrf52_pca20037/led_state_def.h]: Added in led_state.h using as defined value

File: applications/nrf_desktop/configuration/common/led_state.h [hunk 3]
```diff
@@ -47,6 +47,17 @@ enum led_id {
 
 #define LED_UNAVAILABLE CONFIG_DESKTOP_LED_COUNT
 
+#if (defined(CONFIG_BT_PERIPHERAL) && defined(CONFIG_BT_CENTRAL)) || \
+    ((!defined(CONFIG_BT_PERIPHERAL)) && (!defined(CONFIG_BT_CENTRAL)))
+#error Device should be either Bluetooth peripheral or central
+#endif
+
+#ifdef BT_PERIPHERAL
```
Comment (pdunaj) [inline applications/nrf_desktop/configuration/common/led_state.h]: defined(CONFIG_BT_PERIPHERAL)

File: applications/nrf_desktop/configuration/common/led_state.h [hunk 4]
```diff
@@ -47,6 +47,17 @@ enum led_id {
 
 #define LED_UNAVAILABLE CONFIG_DESKTOP_LED_COUNT
 
+#if (defined(CONFIG_BT_PERIPHERAL) && defined(CONFIG_BT_CENTRAL)) || \
+    ((!defined(CONFIG_BT_PERIPHERAL)) && (!defined(CONFIG_BT_CENTRAL)))
```
Comment (pdunaj) [inline applications/nrf_desktop/configuration/common/led_state.h]: some brackets are excessive and do not add to overall readability

File: applications/nrf_desktop/configuration/common/led_state.h (hunk 3 above)
Comment (MarekPieta) [reply applications/nrf_desktop/configuration/common/led_state.h]: done

File: applications/nrf_desktop/configuration/common/led_state.h (hunk 4 above)
Comment (MarekPieta) [reply applications/nrf_desktop/configuration/common/led_state.h]: ok, removed

**Pull request merged**